src/ansys/acp/core/_tree_objects/model.py

```diff
@@ -423,6 +459,149 @@ def export_analysis_model(self, path: _PATH) -> None:
+                    path=path_to_str_checked(path),
+                    remove_midside_nodes=remove_midside_nodes,
+                    layup_representation_3d=layup_representation_3d,
+                    offset_type=offset_type_to_pb(offset_type),  # type: ignore
+                    ascii_encoding=ascii_encoding,
+                    all_elements=all_elements,
+                    element_sets=[element_set._resource_path for element_set in element_sets],
+                    all_plies=all_plies,
+                    plies=[ply._resource_path for ply in plies],
+                )
+            )
+
+    @supported_since("25.1")
+    def import_hdf5_composite_cae(
+        self,
+        path: _PATH,
+        import_mode: HDF5CompositeCAEImportMode = HDF5CompositeCAEImportMode.APPEND,  # type: ignore
+        projection_mode: HDF5CompositeCAEProjectionMode = HDF5CompositeCAEProjectionMode.SHELL,  # type: ignore
+        minimum_angle_tolerance: float = 0.001,
+        recompute_reference_directions: bool = False,
+        shell_mapping_properties: ShellMappingProperties = ShellMappingProperties(),
+        solid_mapping_properties: SolidMappingProperties = SolidMappingProperties(),
+        coordinate_transformation: CoordinateTransformation = CoordinateTransformation(),
+    ) -> None:
+        """Import the lay-up from an HDF5 Composite CAE file.
+
+        Parameters
+        ----------
+        path :
+            File path.
+        import_mode :
+            In :py:attr:`.HDF5CompositeCAEImportMode.APPEND` mode, the imported objects are
+            appended to existing layup.
+            In :py:attr:`.HDF5CompositeCAEImportMode.OVERWRITE` mode, existing objects in the model
+            with the same name are replaced if possible (not locked).
+        projection_mode :
+            Determines whether loaded plies are mapped onto the reference surface
+            (:py:attr:`.HDF5CompositeCAEProjectionMode.SHELL` mode) or exposed as
+            3D plies (:py:attr:`.HDF5CompositeCAEProjectionMode.SOLID` mode).
+        minimum_angle_tolerance :
+            Minimum angle tolerance for which tabular correction angles for plies are computed.
+        recompute_reference_directions :
+            Whether reference directions should be recomputed from tabular angle data or not.
+        shell_mapping_properties :
+            Properties for mapping to the shell on importing HDF5 Composite CAE files.
+            Used only if ``projection_mode`` is set to ``"shell"``.
+        solid_mapping_properties :
+            Properties for importing HDF5 Composite CAE files as imported plies.
+            Used only if ``projection_mode`` is set to ``"solid"``.
+        coordinate_transformation :
+            Coordinate transformation applied to the imported lay-up.
+        """
+        if projection_mode == HDF5CompositeCAEProjectionMode.SHELL:
+            mapping_properties_kwargs: (
+                dict[str, model_pb2.ShellMappingProperties]
```

Comment:
Is the option (or)  dict[str, model_pb2.SolidMappingProperties] needed?

Reply:
Yes, because below (line 583) the same variable is assigned a `dict[str, model_pb2.SolidMappingProperties]`.

An alternate way to write this (avoiding the explicit type hints) would be

```python
if projection_mode == HDF5CompositeCAEProjectionMode.SHELL:
    shell_mapping_properties_kwargs = {"shell_mapping_properties": ...}
    solid_mapping_properties_kwargs = {}
else:
    shell_mapping_properties_kwargs = {}
    solid_mapping_properties_kwargs = {"solid_mapping_properties": ...}

...
model_pb2.ImportHDF5CompositeCAERequest(
    ...
    **shell_mapping_properties_kwargs,
    **solid_mapping_properties_kwargs
)
```

I'm not sure if that's really nicer, though.